src/languageService/kustoLanguageService.ts

```diff
@@ -950,12 +952,45 @@ class KustoLanguageService implements LanguageService {
         return Promise.as(queryParams);
     }
 
-    getGlobalParams(document: ls.TextDocument): Promise<{ name: string; type: string }[]> {
+    getReferencedGlobalParams(document: ls.TextDocument, cursorOffset: number): Promise<{name: string, type: string}[]> {
         if (!this.isIntellisenseV2()) {
             return Promise.as([]);
         }
 
         const script = this.parseDocumentV2(document);
+        let currentBlock = this.getCurrentCommandV2(script, cursorOffset);
```

Comment:
Less of a comment on the PR and more the general use of this library, how would you get the `referencedGlobalParams` across all code blocks in a query? Is there some way to find those offsets without simply parsing by `\n`?

Reply:
"all code blocks in a query" is not a legal sentence. Blocks are a code editing artifact and kusto server does not have any knowledge of them. The entity that gets sent for kusto for execution is always a single query.
Given this I do not see any scenario where there's value in providing a union of all refernced global params in a document.